dali/operators/random/choice.h

```diff
@@ -0,0 +1,277 @@
+  }
+
+  template <typename Generator>
+  DALI_HOST_DEV T Generate(Generator &st) {
+    return dist_(st);
+  }
+
+  DistType dist_ = {};
+};
+
+template <typename Backend>
+class Choice : public rng::RNGBase<Backend, Choice<Backend>, false> {
+ public:
+  using BaseImpl = rng::RNGBase<Backend, Choice<Backend>, false>;
+
+  template <typename T, bool uniform, int dim>
+  using Impl = ChoiceSampleDist<T, uniform, dim>;
+
+  explicit Choice(const OpSpec &spec) : BaseImpl(spec), p_dist_("p", spec) {}
+
+  /**
+   * @brief Returns the number of elements that will be sampled for given sample_idx within batch.
+   *
+   * If the input is scalar, it represents the number of elements, otherwise it's the outermost
+   * dimension (we treat the input as flat list of elements for sampling).
+   */
+  int64_t GetSampleNumElements(const TensorList<CPUBackend> &input, int sample_idx) const {
+    if (input.sample_dim() == 0) {
+      TYPE_SWITCH(input.type(), type2id, T, (DALI_CHOICE_0D_TYPES),
+      (
+        return input.tensor<T>(sample_idx)[0];
+      ),  // NOLINT
+      ());
+    }
+    return input.tensor_shape_span(sample_idx)[0];
+  }
+
+  void AcquireArgs(const OpSpec &spec, const Workspace &ws, int nsamples) {
+    const auto &input = ws.Input<CPUBackend>(0);
+    input_list_shape_.resize(nsamples);
+    for (int sample_idx = 0; sample_idx < nsamples; sample_idx++) {
+      int64_t num_elements = GetSampleNumElements(ws.Input<CPUBackend>(0), sample_idx);
+      DALI_ENFORCE(num_elements > 0,
+                   make_string("Expected positive number of elements for sampling, got: ",
+                               num_elements, " for sample: ", sample_idx, "."));
+      input_list_shape_.set_tensor_shape(sample_idx, {num_elements});
+    }
+    if (p_dist_.HasValue()) {
+      p_dist_.Acquire(spec, ws, nsamples, input_list_shape_);
+      for (int sample_idx = 0; sample_idx < nsamples; sample_idx++) {
+        double sum = 0.0;
+        for (int i = 0; i < p_dist_[sample_idx].num_elements(); i++) {
+          DALI_ENFORCE(p_dist_[sample_idx].data[i] >= 0.0 && p_dist_[sample_idx].data[i] <= 1.0,
+                       make_string("Probabilities must be in range [0, 1], but got: ",
+                                   p_dist_[sample_idx].data[i], " for sample: ", sample_idx,
+                                   " at index ", i, "."));
+          sum += p_dist_[sample_idx].data[i];
+        }
+        DALI_ENFORCE(std::fabs(sum - 1.0) < 1e-4,
+                     make_string("Sum of probabilities must be 1.0, but got ", sum,
+                                 " for sample: ", sample_idx, "."));
+      }
+    }
+  }
+
+  DALIDataType DefaultDataType(const OpSpec &spec, const Workspace &ws) const {
+    if (ws.Input<CPUBackend>(0).sample_dim() == 0) {
+      return DALI_INT32;
+    } else {
+      return ws.Input<CPUBackend>(0).type();
+    }
+  }
+
+  /**
+   * @brief We only allow sampling of the input elements or generating integral types if the
```

Comment:
Nitpick: I'd rather fill @ brief with a brief description of what this function generally does, and add that remark below

Reply:
done. I was thinking about the comment specified in the base class.